[map] always turn on dynamo for map

ydwu4 opened the pull request:
Summary:
X-link: https://github.com/pytorch/executorch/pull/10409

Reland D72896450

Make map consistent with other control flow ops. After the change, map is able to support accessing closures in the map fn.

Test Plan: See existing tests.

Reviewed By: zou3519

Differential Revision: D73138427




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

Skylion007 commented on torch/_higher_order_ops/map.py:
```diff
@@ -125,12 +128,56 @@ def joint_f(*example_args):
+def map(
+    f: Callable[[pytree.PyTree, tuple[pytree.PyTree, ...]], pytree.PyTree],
+    xs: Union[pytree.PyTree, torch.Tensor],
+    *args,
```
Use a TypeVarTuple for args?

zou3519 reviewed (approved):
seems reasonable but tests are failing

ydwu4 committed:
[map] always turn on dynamo for map (#150962) (#152041)

Summary:


Reland D72896450

Make map consistent with other control flow ops. After the change, map is able to support accessing closures in the map fn.

Test Plan: See existing tests.

Reviewed By: zou3519, angelayi

Differential Revision: D73138427

huydhn commented:
@pytorchbot merge -f 'Already land internally'

huydhn commented:
@ydwu4 I think this change is breaking ExecuTorch test in trunk.  Is this something you plan to forward fix?

backends/xnnpack/test/ops/test_conv1d.py::TestConv1d::test_qs8_conv1d_batchnorm_seq [GH job link](https://github.com/pytorch/pytorch/actions/runs/14962302393/job/42026509866) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/ceb009baee09e5c51c5945e94bd5c69166522c74)

ydwu4 commented:
hmm.. I checked the test, the error seems not related to this PR (like it's not using map operator). The abs difference is 0.0220602098852396, which is pretty close to atol is 0.02. Could it be because of instability of the test?
```
	Given atol: 0.02, rtol: 0.02.
	Output tensor shape: torch.Size([2, 2, 4]), dtype: torch.float32
	Difference: max: 0.0, abs: 0.0220602098852396, mean abs error: 0.001378763117827475.
```

huydhn commented:
The failure looks flaky in trunk https://hud.pytorch.org/hud/pytorch/pytorch/main/1?per_page=50&name_filter=executorch, so let me raise this to ExecuTorch team
